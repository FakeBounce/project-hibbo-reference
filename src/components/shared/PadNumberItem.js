import React from 'react';
import {
  number,
  func,
  string,
  bool,
  object,
  array,
  oneOfType,
} from 'prop-types';
import { Text, Image, View } from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';

import underscoreFull from 'assets/padnumber/underscoreFull.png';
import TouchableRipple from 'shared/TouchableRipple';

const styles = EStyleSheet.create({
  $totalMargin: '2 * $size.cardMargin',
  $initialWidth: '100% * 0.33',
  keyboardButton: {
    padding: 8,
    width: '$initialWidth - $totalMargin',
    textAlign: 'center',
    fontSize: 36,
    color: '$colors.primary',
    fontFamily: '$fonts.circularBook',
  },
  img: {
    width: '$initialWidth - $totalMargin',
  },
  containerRipple: {
    alignItems: 'center',
    justifyContent: 'center',
    width: '$initialWidth - $totalMargin',
    height: 63,
  },
});

const PadNumberItem = ({
  parseInput,
  image,
  removeUnderScore,
  value,
  imgStyle,
  visible,
}) =>
  visible ? (
    <TouchableRipple
      styleTouchable={styles.containerRipple}
      onPress={parseInput}
      value={value}
      delay={0}
    >
      {image ? (
        <Image source={image} style={imgStyle} />
      ) : (
        <Text style={styles.keyboardButton}>{value}</Text>
      )}
      {!removeUnderScore && (
        <Image style={styles.img} source={underscoreFull} />
      )}
    </TouchableRipple>
  ) : (
    <View style={styles.containerRipple} />
  );

PadNumberItem.defaultProps = {
  image: undefined,
  removeUnderScore: false,
  imgStyle: {},
  visible: true,
};

PadNumberItem.propTypes = {
  parseInput: func.isRequired,
  image: number,
  removeUnderScore: bool,
  value: string.isRequired,
  imgStyle: oneOfType([object, number, array]),
  visible: bool,
};

export default PadNumberItem;
