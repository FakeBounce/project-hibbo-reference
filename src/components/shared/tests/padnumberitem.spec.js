import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import PadNumberItem from 'shared/PadNumberItem';
import erasePic from 'assets/padnumber/erase.png';

describe('PadNumberItem', () => {
  it('should render the component without image and with underscore', () => {
    const value = '1';
    const parseInput = jest.fn();

    const padNumberItem = renderer
      .create(<PadNumberItem parseInput={parseInput} value={value} />)
      .toJSON();

    expect(padNumberItem).toMatchSnapshot();
  });

  it('should render the component with image and with underscore', () => {
    const value = '1';
    const parseInput = jest.fn();

    const padNumberItem = renderer
      .create(
        <PadNumberItem
          parseInput={parseInput}
          value={value}
          image={erasePic}
        />,
      )
      .toJSON();

    expect(padNumberItem).toMatchSnapshot();
  });

  it('should render the component without image and without underscore', () => {
    const value = '0';
    const parseInput = jest.fn();

    const padNumberItem = renderer
      .create(
        <PadNumberItem
          parseInput={parseInput}
          value={value}
          removeUnderScore
        />,
      )
      .toJSON();

    expect(padNumberItem).toMatchSnapshot();
  });

  it('should render the component with image and without underscore', () => {
    const value = '0';
    const parseInput = jest.fn();

    const padNumberItem = renderer
      .create(
        <PadNumberItem
          parseInput={parseInput}
          value={value}
          image={erasePic}
          removeUnderScore
        />,
      )
      .toJSON();

    expect(padNumberItem).toMatchSnapshot();
  });

  it('should click the component', () => {
    const value = '0';
    const parseInput = jest.fn();

    const tree = shallow(
      <PadNumberItem
        parseInput={parseInput}
        value={value}
        image={erasePic}
        removeUnderScore
      />,
    );

    tree
      .find('TouchableRipple')
      .first()
      .simulate('press');
    expect(parseInput).toHaveBeenCalled();
  });
});
