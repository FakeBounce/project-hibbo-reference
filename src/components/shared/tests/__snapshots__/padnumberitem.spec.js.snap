// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PadNumberItem should render the component with image and with underscore 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={Object {}}
  testID={undefined}
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Image
      source={1}
      style={Object {}}
    />
    <Image
      source={1}
      style={undefined}
    />
  </View>
  <View
    style={
      Array [
        undefined,
      ]
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#303030",
          "borderRadius": 70,
          "height": 70,
          "opacity": 0,
          "transform": Array [
            Object {
              "scale": 0.4,
            },
          ],
          "width": 70,
        }
      }
    />
  </View>
</View>
`;

exports[`PadNumberItem should render the component with image and without underscore 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={Object {}}
  testID={undefined}
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Image
      source={1}
      style={Object {}}
    />
  </View>
  <View
    style={
      Array [
        undefined,
      ]
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#303030",
          "borderRadius": 70,
          "height": 70,
          "opacity": 0,
          "transform": Array [
            Object {
              "scale": 0.4,
            },
          ],
          "width": 70,
        }
      }
    />
  </View>
</View>
`;

exports[`PadNumberItem should render the component without image and with underscore 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={Object {}}
  testID={undefined}
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={undefined}
    >
      1
    </Text>
    <Image
      source={1}
      style={undefined}
    />
  </View>
  <View
    style={
      Array [
        undefined,
      ]
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#303030",
          "borderRadius": 70,
          "height": 70,
          "opacity": 0,
          "transform": Array [
            Object {
              "scale": 0.4,
            },
          ],
          "width": 70,
        }
      }
    />
  </View>
</View>
`;

exports[`PadNumberItem should render the component without image and without underscore 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={Object {}}
  testID={undefined}
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={undefined}
    >
      0
    </Text>
  </View>
  <View
    style={
      Array [
        undefined,
      ]
    }
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#303030",
          "borderRadius": 70,
          "height": 70,
          "opacity": 0,
          "transform": Array [
            Object {
              "scale": 0.4,
            },
          ],
          "width": 70,
        }
      }
    />
  </View>
</View>
`;
